TramsDataApi/UseCases/SearchAcademyTransferProjects.cs

```diff
@@ -3,48 +3,103 @@
-            _indexAcademyTransfer = indexAcademyTransfer;
-        }
-
-        public Task<PagedResult<AcademyTransferProjectSummaryResponse>> Execute(int page, int count, int? urn, string title)
-        {
-            IEnumerable<AcademyTransferProjects> academyTransferProjects = _academyTransferProjectGateway.GetAcademyTransferProjects();
-            
-            IEnumerable<AcademyTransferProjectSummaryResponse> projects = academyTransferProjects.Select(project => _indexAcademyTransfer.AcademyTransferProjectSummaryResponse(project)); //.ToList();
-
-            projects = FilterByUrn(urn, projects);
-            projects = FilterByIncomingTrust(title, projects);
-            int recordTotal = academyTransferProjects.Count();
-            projects = projects.OrderByDescending(atp => atp.ProjectUrn)
-                .Skip((page - 1) * 10).Take(10).ToList();
-            return Task.FromResult(new PagedResult<AcademyTransferProjectSummaryResponse>(projects, recordTotal));
-        }
-        private static IEnumerable<AcademyTransferProjectSummaryResponse> FilterByUrn(int? urn, IEnumerable<AcademyTransferProjectSummaryResponse> queryable)
-        {
-            if (urn.HasValue) queryable = queryable.Where(p => p.ProjectUrn == urn.ToString()); //.ToList();
-
-            return queryable;
-        }
-        private static IEnumerable<AcademyTransferProjectSummaryResponse> FilterByIncomingTrust(string title, IEnumerable<AcademyTransferProjectSummaryResponse> queryable)
-        {
-            if (!string.IsNullOrWhiteSpace(title))
+   public class SearchAcademyTransferProjects : ISearchAcademyTransferProjects
+   {
+      private readonly IAcademyTransferProjectGateway _academyTransferProjectGateway;
+      private readonly ITrustGateway _trustGateway;
+
+      public SearchAcademyTransferProjects(
+         IAcademyTransferProjectGateway academyTransferProjectGateway,
+         ITrustGateway trustGateway)
+      {
+         _academyTransferProjectGateway = academyTransferProjectGateway;
+         _trustGateway = trustGateway;
+      }
+
+      public Task<PagedResult<AcademyTransferProjectSummaryResponse>> Execute(int page, int count, int? urn,
+         string title)
+      {
+         IEnumerable<AcademyTransferProjects> academyTransferProjects = FilterByUrn(
+            _academyTransferProjectGateway.GetAcademyTransferProjects(), urn).ToList();
+
+         IEnumerable<AcademyTransferProjectSummaryResponse> projects =
+            FilterByIncomingTrust(title, AcademyTransferProjectSummaryResponse(academyTransferProjects));
+
+         var recordTotal = projects.Count();
+         projects = projects.OrderByDescending(atp => atp.ProjectUrn)
+            .Skip((page - 1) * 10).Take(10).ToList();
+
+         return Task.FromResult(new PagedResult<AcademyTransferProjectSummaryResponse>(projects, recordTotal));
+      }
+
+      private static IEnumerable<AcademyTransferProjects> FilterByUrn(IEnumerable<AcademyTransferProjects> queryable,
```

Comment:
Just a thought - you could return this as a dictionary. I think Linq offers a ToDictionary extension.

Reply:
Fair enough.